include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -85,6 +85,208 @@ DEAL_II_NAMESPACE_OPEN
+            ExcDimensionMismatch (vec.size(), dof_handler.n_dofs()));
+
+    Assert (dof_handler.get_fe().n_components() == function.n_components,
+            ExcDimensionMismatch(dof_handler.get_fe().n_components(), function.n_components));
+
+    Assert (component_mask.n_selected_components(dof_handler.get_fe().n_components()) > 0,
+            ComponentMask::ExcNoComponentSelected());
+    //
+    // Computing the generalized interpolant isn't quite as straightforward
+    // as for classical Lagrange elements. A major complication is the fact
+    // it generally doesn't hold true that a function evaluates to the same
+    // dof coefficient on different cells. This means *setting* the value
+    // of a (global) degree of freedom computed on one cell doesn't
+    // necessarily lead to the same result when computed on a neighboring
+    // cell (that shares the same global degree of freedom).
+    //
+    // We thus, do the following operation:
+    // For every degree of freedom that is shared by k cells, we compute
+    // its value on all k cells and take the weighted average with respect
+    // to the JxW values.
+    //
+    // FIXME Write rest of documentation.
+    //
+    // It can be shown that this procedure leads is a projection...
+    //
+
+    typedef typename VectorType::value_type number;
+
+    const hp::FECollection<dim, spacedim> fe(dof_handler.get_fe());
+
+    std::vector<types::global_dof_index> dofs_on_cell(fe.max_dofs_per_cell());
+
+    // Temporary storage for cell-wise interpolation operation. We store a
+    // variant for every fe we encounter to speed up resizing operations.
+    // The first vector is used for local function evaluation. The vectors
+    // dof_values and dof_values_weights are used to store intermediate
+    // cell-wise interpolation results (see the detailed explanation in the
+    // for loop further down below).
+
+    std::vector<std::vector<Vector<number> > > fe_function_values(fe.size());
+    std::vector<std::vector<number> > fe_dof_values(fe.size());
+    std::vector<std::vector<number> > fe_dof_values_weights(fe.size());
+
+    // We will need two temporary global vectors that store the new values
+    // and weights.
+    VectorType interpolation;
+    VectorType weights;
+    interpolation.reinit(vec);
+    weights.reinit(vec);
+
+    // We use an FEValues object to transform all generalized support
+    // points from the unit cell to the real cell coordinates. Thus,
+    // initialize a quadrature with all generalized support points and
+    // create an FEValues object with it.
+
+    hp::QCollection<dim> support_quadrature;
+    for (unsigned int fe_index=0; fe_index<fe.size(); ++fe_index)
+      {
+        const auto &points = fe[fe_index].get_generalized_support_points();
+        const auto size = points.size();
+        std::vector<double> weights(size, 1. / size);
+        support_quadrature.push_back(Quadrature<dim>(points, weights));
```

Comment:
since you don't need the weights, just create the quadrature with only the points -- there is a constructor for that.

Reply:
@bangerth I need valid JxW values (even though the weight doesn't matter). Unfortunately, said constructor initializes the weights with infinity. This took me 2h to figure out yesterday.

Reply:
Hm. What do you need the JxW values for? If the weight doesn't matter, isn't all you need the determinant of the Jacobian?

Reply:
Good point. I have updated the code.